src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1054,12 +1054,31 @@ private void DoGraphAutoLayout(object o)
         {
             if (Model.Nodes.Count() == 0)
                 return;
+            
+            var selection = DynamoSelection.Instance.Selection;
+
+            bool isInternalLayout = selection.Count > 0 &&
+                selection.All(x => x is AnnotationModel ||
+                selection.Where(g => g is AnnotationModel)
+                .Any(g => (g as AnnotationModel).SelectedModels.Contains(x)));
+            
+            if (!isInternalLayout)
+            {
+                GenerateCombinedGraph();
+            }
+            else
+            {
+                GenerateCombinedGroupGraph();
+            }
```

Comment:
So `GenerateCombinedGraph` is meant for selections and ungrouped nodes in graph and `GenerateCombinedGroupGraph` is meant for groups only? Am I right? If so what happens if we make a selection of both group and some ungrouped nodes in the graph or is that something you're not supporting as of now?